frontend/src/views/MigrationHistoryDetail.vue

```diff
@@ -199,21 +252,43 @@ export default defineComponent({
   setup(props) {
     const store = useStore();
 
-    const migrationHistory = computed((): MigrationHistory => {
-      return store.getters["instance/migrationHistoryById"](
-        idFromSlug(props.migrationHistorySlug)
+    const database = computed((): Database => {
+      return store.getters["database/databaseById"](
+        idFromSlug(props.databaseSlug)
       );
     });
 
+    // get all migration histories before (include) the one of given id, ordered by descending version.
+    const prevMigrationHistoryList = computed((): MigrationHistory[] => {
+      const migrationHistoryList: MigrationHistory[] = store.getters[
+        "instance/migrationHistoryListByInstanceIdAndDatabaseName"
+      ](database.value.instance.id, database.value.name);
+      return migrationHistoryList.filter(
+        (mh) => mh.id <= idFromSlug(props.migrationHistorySlug)
+      );
+    });
+
+    const migrationHistory = computed(
+      (): MigrationHistory => prevMigrationHistoryList.value[0]
+    );
+
+    // previousHistorySchema is the schema snapshot of the last migration history before the one of given id.
+    const previousHistorySchema = computed(
+      (): string => prevMigrationHistoryList.value[1].schema
```

Comment:
How do we handle when length is 1 or 0? Same for migrationHistory.

Reply:
done.
when length is 0, query the given id.
previous schema is only used when length > 1